wagtail/wagtailusers/templates/wagtailusers/users/edit.html

```diff
@@ -27,15 +27,20 @@
                     {% include "wagtailadmin/shared/field_as_li.html" with field=form.password1 %}
                     {% include "wagtailadmin/shared/field_as_li.html" with field=form.password2 %}
                     {% include "wagtailadmin/shared/field_as_li.html" with field=form.is_active %}
-                    
-                    <li><input type="submit" value="{% trans 'Save' %}" /></li>
+
+                    <li>
+                        <input type="submit" value="{% trans 'Save' %}" />
+                        <a href="{% url 'wagtailusers_users:delete' user.id %}" class="button button-secondary no">{% trans "Delete user" %}</a>
```

Comment:
We should have a permission check around this button, so that it doesn't get shown to users who have edit but not delete permission.